src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -97,8 +93,16 @@ public PooledQueueCache(ICacheDataAdapter<TQueueMessage, TCachedMessage> cacheDa
             this.cacheDataAdapter = cacheDataAdapter;
             this.comparer = comparer;
             this.logger = logger.GetSubLogger("messagecache", "-");
+            this.itemCount = 0;
             pool = new CachedMessagePool<TQueueMessage, TCachedMessage>(cacheDataAdapter);
             messageBlocks = new LinkedList<CachedMessageBlock<TCachedMessage>>();
+            this.cacheMonitor = cacheMonitor;
+
+            if (this.cacheMonitor != null && cacheMonitorWriteInterval.HasValue)
+            {
+                var timer = new Timer(this.ReportCacheMessageStatistics, null, cacheMonitorWriteInterval.Value, cacheMonitorWriteInterval.Value);
```

Comment:
?? I'm confused by your text. I understand and agree that I should dispose it when receiver is shutting down. What does the text before that mean?  I think it is good if the timer is GCed when the obejct is GCed, no?